docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+    contiguous sequence of bytes. This specification defines two types
+    of memory layout based on "C" (row-major) and "F" (column-major)
+    ordering of elements_, but extensions may define other
+    memory layouts.
+
+.. _compressor:
+.. _compressors:
+
+*Compressor*
+
+    An array_ may be associated with a compressor, which is a codec_
+    that transforms the binary representation of a chunk_ in some way,
+    usually to reduce data size.
+
+.. _codec:
+.. _codecs:
+
+*Codec*
+
+    A codec is a pair of algorithms which transform binary data in
+    some way and are used to encode and decode chunks_. This
+    specification defines a codec interface which comprises
+    a pair of operations, one to perform the transformation (encode),
+    the other to reverse the transformation (decode). This
+    specification only considers the case where a codec is used as a
+    compressor_, but extensions may extend the chunk encoding
+    process, for example, to add support for one or more filter codecs
+    to be applied prior to compression.
+
+.. _metadata document:
+.. _metadata documents:
+
+*Metadata document*
+
+    Each array_ in a hierarchy_ is represented by a metadata document,
+    which is a machine-readable document containing essential
+    processing information about the node. For example, an array_
+    metadata document will specify the number of dimensions_, shape_,
+    `data type`_, grid_, `memory layout`_ and compressor_ for that
+    array_.
+
+    Groups can have a optional metadata document which provide extra
+    information about a group.
+
+.. _store:
+.. _stores:
+
+*Store*
+
+    The `metadata documents`_ and encoded chunk_ data for all nodes in
+    a hierarchy_ are held in a store. To enable a variety of different
+    store types to be used, this specification defines an `Abstract store
+    interface`_ which is a common set of operations that stores may
+    provide.
+
+.. _storage transformer:
+.. _storage transformers:
+
+*Storage transformer*
+
+    To enhance the storage capabilities, storage transformers may
+    change the storage structure and behaviour of data coming from
+    an array_ in the underlying store_. Upon retrieval the original data
+    is restored within the transformer. Any number of `predefined storage
+    transformers`_ can be registered and stacked.
+    See the `storage transformers details`_ below.
+
+.. _`storage transformers details`: #storage-transformers-1
+
+Node names
+==========
+
+Except for the root node, each node in a hierarchy must have a name,
+which is a string of characters. To ensure consistent behaviour
+across different storage systems, the following constraints apply to
+node names:
+
+* must not be the empty string ("")
+
+* must use only characters in the sets ``a-z``, ``A-Z``, ``0-9``,
+  ``-_.``
+
+* must not be a string composed only of period characters, e.g. "." or
+  ".."
+
+* must be at most 255 characters long
```

Comment:
This sounds a bit restrictive to me. Often auto-generated names can get longer. Is there a technical reason to prohibit this? (E.g. incompatibility with some target storage format.) Otherwise this could be a 'should'.

Reply:
https://support.code42.com/Incydr/Agent/Troubleshooting/Windows_file_paths_longer_than_255_characters

Reply:
I think it could make sense to include this (or ideally a more authoritative reference from microsoft itself) as a note, but I think as with the case sensitivity issue we shouldn't necessarily put that as a requirement in the spec, but instead just note possible interoperability issues.

Reply:
Adapting this in PR #175, see 882e466. Marking this as resolved, please feel free to comment in the linked PR directly.